### PR TITLE
[SIMULATION] fixes needed to build cxxmodules

### PR DESCRIPTION
**Pull request by smuzaffar**
#### PR description:

- Added missing include to avoid clang error about unknown type `uint32_t`

### CODE REVIEW

**Comment by smuzaffar** [comment]
please test

**Comment by civanch** [comment]
+1

File: DataFormats/HcalDetId/interface/HcalTestNumbering.h [hunk 1]
```diff
@@ -6,6 +6,7 @@
 ///////////////////////////////////////////////////////////////////////////////
 
 #include <boost/cstdint.hpp>
+#include <stdint.h>
```
Comment (kpedro88) [inline DataFormats/HcalDetId/interface/HcalTestNumbering.h]: usually we recommend using `<c*>` over `<*.h>` headers for STL.

In addition, this file already includes `<boost/cstdint.hpp>`, which seems to define `uint32_t`. I am confused why the specific added include is necessary.

Comment (smuzaffar) [reply DataFormats/HcalDetId/interface/HcalTestNumbering.h]: @kpedro88 , when CXX Modules are generated (using clang) then clang complains about unknown type `uint32_t`

Comment (fabiocos) [reply DataFormats/HcalDetId/interface/HcalTestNumbering.h]: @smuzaffar ok, but is this behaviour related to the use of <*.h> vs <c*> ? I would naively expect it is not...

Comment (civanch) [reply DataFormats/HcalDetId/interface/HcalTestNumbering.h]: @smuzaffar , we have in many other places

#include <cstdint>

likely it is fully equivalent, do we have a guideline what to use?

Comment (smuzaffar) [reply DataFormats/HcalDetId/interface/HcalTestNumbering.h]: I will try cstdint and update the PR if works

Comment (fabiocos) [reply DataFormats/HcalDetId/interface/HcalTestNumbering.h]: @civanch I do not think we have any official written guideline, I think this is more a habit, @smuzaffar @davidlange6 do you confirm?

**Comment by smuzaffar** [comment]
please test

File: DataFormats/HcalDetId/interface/HcalTestNumbering.h (hunk 1 above)
Comment (smuzaffar) [reply DataFormats/HcalDetId/interface/HcalTestNumbering.h]: fixed, it was enough to releance boost/cstdint with csdtint

**Comment by civanch** [comment]
+1

**Comment by fabiocos** [comment]
@smuzaffar thanks!

**Comment by fabiocos** [comment]
+1

**Pull request merged**